src/crypto/DefaultSessionKeystore.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#if CHIP_HAVE_CONFIG_H
+#include <crypto/CryptoBuildConfig.h>
+#endif
+
+#if CHIP_CRYPTO_PSA
+#include <crypto/PSASessionKeystore.h>
+#else
+#include <crypto/RawKeySessionKeystore.h>
+#endif
+
+namespace chip {
+namespace Crypto {
+
+inline SessionKeystore & GetDefaultSessionKeystore()
+{
+#if CHIP_CRYPTO_PSA
+    static PSASessionKeystore sKeystore;
+#else
+    static RawKeySessionKeystore sKeystore;
+#endif
+
+    return sKeystore;
+}
```

Comment:
Why make this singleton/global? This means we cannot have multiple clients in a single process, if there is a server. I would argue that RawSessionKeystore be the default de-facto in Server.h for the **now problematic** CommonCaseServerInitParams, and let the clients change the Keystore to be the kind they need. This change here significantly impacts the testability/usability of multiple nodes in same process, at least at the crypto later...

Reply:
The session keystore interface is coupled with CHIPCryptoPAL so if the underlying crypto operations expect key references, you cannot use `RawKeySessionKeystore`. This is a static local so if your code never calls this function the object will not be included in the binary.